tests/unit/components/popup-dialog/PopupDialog.spec.jsx

```diff
@@ -0,0 +1,43 @@
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import PopupDialog from '../../../../src/components/popup-dialog/PopupDialog'
+import { vi } from 'vitest'
+
+describe('PopupDialog component', () => {
+  const props = {
+    content: 'TestContent',
+    closeModal: vi.fn(),
+    paperProps: {},
+    timerId: 2000,
+    closeModalAfterDelay: vi.fn()
+  }
+  vi.useFakeTimers()
+
+  beforeEach(() => {
+    render(<PopupDialog {...props} />)
+  })
+
+  it('Component renders correctly', () => {
+    const contentElement = screen.getByText('TestContent')
+    expect(contentElement).toBeInTheDocument()
+  })
+
+  it('Dialog closes when you click on the close button', async () => {
```

Comment:
There is nothing asynchronous inside, so you don't need `async/await`. `userEvent.click()` is synchronous.

Reply:
This click starts an asynchronous function inside the component, I can't spy this function, without asynchrony in this test.